### PR TITLE
fix cli options

### PR DESCRIPTION
**Pull request by jianyexi**
Fix https://github.com/Azure/cadl-azure/issues/1001 , we need to have separate path for service code

### CODE REVIEW

File: packages/compiler/core/cli.ts [hunk 1]
```diff
@@ -274,10 +279,17 @@ async function getCompilerOptions(args: {
   import?: string[];
   watch?: boolean;
   "diagnostic-level": string;
+  "service-code-path"?: string;
 }): Promise<CompilerOptions> {
   // Ensure output path
   const outputPath = resolve(args["output-path"]);
+  const serviceCodePath = args["service-code-path"]
```
Comment (markcowl) [inline packages/compiler/core/cli.ts:286]: similarly, I think all of this code preparing the directory needs to be in the generator itself

File: packages/compiler/core/cli.ts [hunk 2]
```diff
@@ -66,6 +66,11 @@ async function main() {
             default: false,
             describe: "Watch project files for changes and recompile.",
           })
+          .option("service-code-path", {
```
[the remark this answers is not in the document]
Comment (markcowl) [reply packages/compiler/core/cli.ts:69]: Agreed.  does this show  up as ```program.compilerOptions.miscOptions.serviceCodePath```?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/compiler/core/cli.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jianyexi) [reply packages/compiler/core/cli.ts:69]: So we need to remove program.compilerOptions.serviceCodePath,  as I used to consider it's a common option for all the service codes in the future

[the remark this answers is not in the document]
Comment (markcowl) [reply packages/compiler/core/cli.ts:69]: Yes. the extensibility model for cadl models things this way, so we should use this facility - we may have special case clis in the future when we refactor emitters, and these may have different command line options, bot for now, using the emitter-defined extensible options is the right thing.

**Comment by markcowl** [comment]
Closing this PR, as I think all the necessary changes are in the emitter.

**Pull request closed without merging**